### ISSUE TITLE
Support for Python 3.x

### ISSUE THREAD

**Issue by Talvalin** [tableau/VizAlerts#111]
Current documentation states that Python 2.7 is required, although it seems that most of the dependencies have Python 3 compatible versions, so is it possible to use VizAlerts with Python 3 now, or is some additional work required?

**Comment by mcoles** [comment]
We've simply never tested with Python 3, and it never made a big enough priority to get attention. If you want to test on Python 3 and make code changes so that it is compatible, that'd be awesome. I suspect that there wouldn't be too many changes required.

**Comment by t8y8** [comment]
I recommend running `futurize` over it and examining the output

**Comment by mcoles** [comment]
I'm working my way through this now that Python 2 is no longer supported. Frankly, it's been much harder than I would have thought. There are lots of differences in encoding handling between the two versions, and lots of the code that I had to use in Python 2 to properly encode in UTF-8 is breaking in Python 3, or requires additional explicit calls to .encode() to get working. I've got it so I can actually process alerts now, but there are still issues properly processing emails with non-ASCII characters, so it's still a work in progress.
